### PR TITLE
Using base node functionality instead of express methods

### PR DESCRIPTION
**Pull request by FredrikFolkesson**
Change to using base Node request and response methods instead of express ones.

This is to enable support of other frameworks (Koa)

Using koa-connect the function will be called with just the base `req` and `res` objects from vanilla node, so only methods on them should be used.

Also enabled JSON metrics if not accepting "text""

This is related to #8

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -123,10 +123,17 @@ class MetricsMiddleware {
 
   metricsRoute(req, res) {
     if (req.headers['x-forwarded-for']) {
-      return res.sendStatus(404);
+      res.writeHead(404, {
+        'Content-Type': 'text/plain',
+      });
+      return res.end('Not Found');
```
Comment (hairyhenderson) [inline index.js]: Is this related to this PR? Not sure why this is a necessary change...

Comment (hairyhenderson) [reply index.js]: Ah, ok 👍

**Comment by hairyhenderson** [comment]
> Also enabled JSON metrics if not accepting "text""

I'd rather that happen in a separate PR. JSON isn't a valid Prometheus exposition format, so I'm not super certain that it's even a good idea. Though I'm not opposed to the idea. I'd just rather keep the PR focused 😉

**Comment by FredrikFolkesson** [comment]
@hairyhenderson we need it for `mira` that uses this library.
